examples/widgets/progress_bar.rs

```diff
@@ -0,0 +1,156 @@
+//! This example illustrates how to setup and use the [`ProgressBarWidget`].
+//! Any Node that has a `ProgressBar` component *and* an immediate child node
+//! with the `ProgressBarInner` component will be considered a ProgressBar-widget.
+
+use bevy::{
+    math::map_range,
+    prelude::*,
+    widget::{ProgressBarInner, ProgressBarWidget, WidgetPlugin},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_plugin(WidgetPlugin)
+        .insert_resource(Progress::Completed(2.0))
+        .add_startup_system(setup)
+        .add_system(update_progress_state)
+        .add_system(set_widget_progress.after(update_progress_state))
+        .add_system(update_widget_text.after(set_widget_progress))
+        .run();
+}
+
+fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
+    let canvas_background: Color = Color::rgba_u8(19, 21, 22, 255);
+    let progress_bar_background: Color = Color::rgba_u8(29, 31, 33, 255);
+    let progress_bar_foreground: Color = Color::rgba_u8(50, 104, 159, 255);
+    let text_color: Color = Color::rgba_u8(197, 198, 190, 255);
+    let font = asset_server.load("fonts/FiraMono-Medium.ttf");
+
+    // ui camera
+    commands.spawn(Camera2dBundle::default());
+
+    // background that fills the entire viewport
```

Comment:
Is this necessary? Why not just use the default clear color of bevy? I'm not really against this, but just trying to see where the example could be simplified a little.

Reply:
I guess you need the root node to take the whole screen to center the progress bar?

Reply:
Yes. Previously we have been using auto-margin to centre elements on the page. This is not the best practice, see #6535. In order to centre the element you need the parent element to cover the whole area. 

The root / background does not strictly need a background colour, I added it for stylistic choices but can remove it if desired.